drivers/include/lsm6ds3.h

```diff
@@ -0,0 +1,221 @@
+    LSM6DS3_ACC_BANDWITH_200HZ = 0x01, /** 200Hz Anti-aliasing */
+    LSM6DS3_ACC_BANDWITH_100HZ = 0x02, /** 100Hz Anti-aliasing */
+    LSM6DS3_ACC_BANDWITH_50HZ  = 0x03, /** 50Hz  Anti-aliasing */
+} lsm6ds3_acc_bandwith_t;
+
+/**
+ * @brief Possible gyroscope sample rates
+ */
+typedef struct {
+    LSM6DS3_GYR_SAMPLE_RATE_13HZ   = 0x01,  /**   13Hz sample rate */
+    LSM6DS3_GYR_SAMPLE_RATE_26HZ   = 0x02,  /**   26Hz sample rate */
+    LSM6DS3_GYR_SAMPLE_RATE_52HZ   = 0x03,  /**   52Hz sample rate */
+    LSM6DS3_GYR_SAMPLE_RATE_104HZ  = 0x04,  /**  104Hz sample rate */
+    LSM6DS3_GYR_SAMPLE_RATE_208HZ  = 0x05,  /**  208Hz sample rate */
+    LSM6DS3_GYR_SAMPLE_RATE_416HZ  = 0x06,  /**  416Hz sample rate */
+    LSM6DS3_GYR_SAMPLE_RATE_833HZ  = 0x07,  /**  833Hz sample rate */
+    LSM6DS3_GYR_SAMPLE_RATE_1660HZ = 0x08,  /** 1660Hz sample rate */
+} lsm6ds3_gyr_sample_rate_t;
+
+/**
+ * @brief Possible gyroscope scales
+ */
+typedef struct {
+    LSM6DS3_GYR_SCALE_245DPS  = 0x00, /**  245 degrees per second */
+    LSM6DS3_GYR_SCALE_500DPS  = 0x01, /**  500 degrees per second */
+    LSM6DS3_GYR_SCALE_1000DPS = 0x02, /** 1000 degrees per second */
+    LSM6DS3_GYR_SCALE_2000DPS = 0x03, /** 2000 degrees per second */
+} lsm6ds3_gyr_scale_t;
+
+/**
+ * @brief Possible gyroscope scale at 125 dps
+ */
+typedef struct {
+    LSM6Ds3_GYR_SCALE_125_DI = 0x00,
+    LSM6DS3_GYR_SCALE_125_EN = 0x01,
+} lsm6ds3_gyr_scale_125_t;
+
+/**
+ * @brief Initialize a new LSM6DS3 device.
+ *
+ * @param[in] dev               device descriptor of an LSM303DLHC device
+ * @param[in] i2c               I2C device the sensor is connected to
+ * @param[in] acc_pin           GPIO pin connected to accelerometer
+ * @param[in] gyr_pin           GPIO pin connected to gyroscope
+ * @param[in] acc_address       I2C address of the accelerometer
+ * @param[in] acc_sample_rate   accelerometer sample rate
+ * @param[in] acc_scale         accelerometer scale (from +- 2g to +- 16g)
+ * @param[in] acc_bandwith      accelerometer bandwith (Anti-aliasing)
+ * @param[in] gyr_sample_rate   gyroscope sample rate
+ * @param[in] gyr_scale         gyroscope scale (degrees per second)
+ * @param[in] gyr_scale_125     gyroscope full-scale at 125 dps
+ *
+ * @return                       0 on success
+ * @return                      -1 on error
+ */
+int lsm6ds3_init(lsm6ds3_t *dev,
+                 i2c i2c,
+                 gpio_t acc_pin,
+                 gpio_t gyr_pin,
+                 uint8_t acc_address,
+                 lsm6ds3_acc_sample_rate_t sample_rate,
+                 lsm6ds3_acc_scale_t acc_scale,
+                 lsm6ds3_acc_bandwith_t acc_bandwith,
+                 uint_8_t gyr_address,
+                 lsm6ds3_gyr_sample_rate_t gyr_sample_rate,
+                 lsm6ds3_gyr_scale_t gyr_scale,
+                 lsm6ds3_gyr_scale_125_t gyr_scale_125);
+
+/**
+ * @brief Read a accelerator value from the sensor.
+ *
+ * @details TODO:
```

Comment:
Okey, I remove it, maybe I put later some details there, when I can test the board.